src/components/confirmation-dialog.js

```diff
@@ -0,0 +1,134 @@
+ * @slot accept - (Optional) Customizes the accept button text. Defaults to "Accept."
+ *
+ * @cssproperty [--dialog-padding=1em] - The padding inside the dialog.
+ * @cssproperty [--dialog-background-color=#242424] - The background color of the dialog.
+ * @cssproperty [--dialog-border=1px solid #666] - The border style for the dialog.
+ * @cssproperty [--dialog-color=rgb(255 255 255 / 87%)] - The text color inside the dialog.
+ * @cssproperty [--dialog-title-font-weight=600] - The font weight for the dialog title.
+ * @cssproperty [--dialog-border-radius=0.5em] - The border radius of the dialog.
+ * @cssproperty [--dialog-max-width=360px] - The maximum width of the dialog.
+ * @cssproperty [--button-color=rgb(255 255 255 / 87%)] - The text color of the buttons.
+ * @cssproperty [--button-border-radius=0.5em] - The border radius of the buttons.
+ * @cssproperty [--accept-button-background=#9D4040] - The background color of the accept button.
+ * @cssproperty [--accept-button-hover-background=#733030] - The background color of the accept button on hover.
+ * @cssproperty [--cancel-button-background=rgb(255 255 255 / 0%)] - The background color of the cancel button.
+ * @cssproperty [--cancel-button-hover-background=rgb(255 255 255 / 20%)] - The background color of the cancel button on hover.
+ * @cssproperty [--cancel-button-hover-color=rgb(0 0 0 / 87%)] - The text color of the cancel button on hover.
+ *
+ * @example
+ * <confirmation-dialog>
+ *   <span slot="title">Are you sure ?</span>
+ *   <span slot="content">This action is irreversible</span>
+ * </confirmation-dialog>
+ */
+class ConfirmationDialog extends LitElement {
+  static styles = unsafeCSS(confirmationDialogStyle);
+
+  static properties = {
+    open: { type: Boolean, state: true },
+    accepted: { type: Boolean, state: true }
+  };
+
+  /**
+   * Reference to the dialog HTML element.
+   *
+   * @type {import('lit/directives/ref').Ref<HTMLDialogElement>}
+   * @private
+   */
+  #dialog = createRef();
+
+  constructor() {
+    super();
+    this.open = false;
+    this.accepted = false;
+  }
+
+  render() {
+    return html`
+      <dialog ${ref(this.#dialog)}>
+        <div part="body">
+          <slot name="title"></slot>
+          <slot name="content"></slot>
+          <div part="actions">
+            <button part="button cancel" @click=${() => this.#close(false)}>
+              <slot name="cancel">Cancel</slot>
+            </button>
+            <button part="button accept" @click=${() => this.#close(true)}>
+              <slot name="accept">Accept</slot>
+            </button>
+          </div>
+        </div>
+      </dialog>
+    `;
+  }
+
+  #close(accepted = false) {
+    this.open = false;
+    this.accepted = accepted;
+  }
+
+  updated(_changedProperties) {
+    super.updated(_changedProperties);
+
+    if (_changedProperties.has('open')) {
+      this.open ? this.#dialog.value.showModal() : this.#dialog.value.close();
```

Comment:
So basically it's an `if` statement hidden in a ternary.

Reply:
Changed, I've move it to `dispatchEvent` function and renamed it to `updateDialog`.